From c0d7e4df23a4b7891baed07e561b894d67ba047a Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Thu, 29 Aug 2019 06:50:05 +0000
Subject: [PATCH 21/81] glamor/glamor_egl.c: Fix memory leak of device_path

The device_path is strdup-ed during screen init, thus we need to free it
in CloseScreen().

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 glamor/glamor_egl.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/glamor/glamor_egl.c b/glamor/glamor_egl.c
index 10813cbd1..096007013 100644
--- a/glamor/glamor_egl.c
+++ b/glamor/glamor_egl.c
@@ -793,6 +793,11 @@ glamor_egl_close_screen(ScreenPtr screen)
 
     glamor_egl_set_pixmap_bo(screen_pixmap, NULL, pixmap_priv->used_modifiers);
 
+    if (glamor_egl->device_path) {
+        free(glamor_egl->device_path);
+        glamor_egl->device_path = NULL;
+    }
+
     screen->CloseScreen = glamor_egl->saved_close_screen;
 
     return screen->CloseScreen(screen);
@@ -921,7 +926,6 @@ static void glamor_egl_cleanup(struct glamor_egl_screen_private *glamor_egl)
     }
     if (glamor_egl->gbm)
         gbm_device_destroy(glamor_egl->gbm);
-    free(glamor_egl->device_path);
     free(glamor_egl);
 }
 
-- 
2.38.5

